### ISSUE TITLE
Register processing script with raw dataset URL

### ISSUE THREAD

**Issue by mdoering** [Sp2000/colplus-backend#533]
For many datasets we still need to convert a custom formats into ColDP/DwC-A/ACEF.
To automise that we could include (bash) scripts or maybe better (docker) containers to process them into one of the 3 formats automatically during each import.

The proxied Paris dataset is actually a special case of this. IPNI, Kew PoWo and many more would benefit from this to be kept up to date automatically.

**Comment by mdoering** [comment]
@gdower is that sth useful to consider?

**Comment by gdower** [comment]
@mdoering, I like the idea of using docker because it would allow greater flexibility, and would be  compatible with past work.

**Comment by mdoering** [comment]
@gdower yes, scripts are highly depending on the surrounding infrastructure. If we would just run a docker container specialised on conversions we could do all sorts of things. And containers could even inherit from more general transformation containers to become more specialised. If you think that a way to go forward let's think about interaction details. We would need a generic way to spin up the concrete container, to execute the processing by passing over the dataset details (datasetKey, accessUrl and format?) and to return the packaged archive.

**Comment by mdoering** [comment]
It's not super urgent, but yes. If you have time please do so and we can look into how I can spin up the containers from java and shut them down when done. Error logging will be an issue. As well as reporting of current state/progress and a final notification that the conversion has been completed or failed. Maybe many of that can go through ELK.

I feel we should keep it simple and spin up new containers for each conversion, shutting them down again after we have finished.

**Comment by gdower** [comment]
We could have a log directory that includes a PID file that the clearinghouse could watch for the status of conversion. The World Plants conversion actually is taking a really long time to run, so even percent done might be useful. (I haven't optimized the World Plants converter for performance yet.)

**Comment by mdoering** [comment]
@gdower didn't you guys have a kubernetes cluster we might be able to use if this proves useful? Could we manage containers remotely or is the cluster not exposed at all?

**Comment by gdower** [comment]
That might be possible. Let me talk with Dima about it.
